porogo/api/models.py

```diff
@@ -1,3 +1,122 @@
+'''Define modules used by the backend'''
 from django.db import models
 
-# Create your models here.
+class URL(models.Model):
+    """
+    Stores URLs for many to many relationships
+    
+    Attributes:
+        url (string): URL to store
+    """
+    url = models.URLField(max_length=200)
+
+def get_dest_path(_, filename):
+    """
+    Get path for a given file based on its filename
+    """
+    filename = filename.replace(" ", "")
+    index_of_dash = filename.find('-')
+
+    if index_of_dash == -1:
+        return f'media/other/{filename}' 
+    return f'media/{filename[None:index_of_dash]}/{filename}'
+
+class Image(models.Model):
+    """
+    Stores URLs for many to many relationships
+    
+    Attributes:
+        url (string): URL to store
+    """
+    img = models.FileField(upload_to=get_dest_path)
+    desc = models.CharField(max_length=32)
+
+class Language(models.Model):
+    """
+    Stores Languages for many to many relationships
+    
+    Attributes:
+        name (string): language name to store
+    """
+    name = models.CharField(max_length=16, unique=True)
+
+class Developer(models.Model):
+    """
+    Stores Developers on the site
+
+    Attributes:
+        first_name (str): First name of dev
+        last_name (str): Last name of dev
+        email (str): Email of dev
+        short_description (str): Description of the dev
+        description_experience (str): Description of the dev's experience
+        description_personal (str): Description of the dev's personal interests
+        github_name (str): Github username of dev
+        linkedin_url (str): Linkedin url for the dev
+        other_url (str): Url to another site
+        other_url_type (str): Type of the other url
+        picture (File): Profile picture of the dev
+        featured_project (Project): Project that the user wants to display
```

Comment:
foreignkey

Reply:
Foreign key give you a Project object when you access it in code so we wouldnt need to care that its a FK when using it. Read about it more [here](https://docs.djangoproject.com/en/5.1/topics/db/examples/many_to_one/)